lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -82,17 +89,7 @@ def render(engine, data, *)
               end
             end
 
-            tracer = Datadog.configuration[:sinatra][:tracer]
-            distributed_tracing = Datadog.configuration[:sinatra][:distributed_tracing]
-
-            if distributed_tracing && tracer.provider.context.trace_id.nil?
-              context = HTTPPropagator.extract(request.env)
-              tracer.provider.context = context if context.trace_id
-            end
-
-            span = tracer.trace('sinatra.request',
-                                service: Datadog.configuration[:sinatra][:service_name],
-                                span_type: Datadog::Ext::HTTP::TYPE)
+            span = RequestSpanMiddleware.fetch_sinatra_trace(request.env)
```

Comment:
Below theres a `@datadog_request_span`. Is that obsolete? Should it just be using this middleware to drive span storage/retrieval?

Reply:
It seems I missed your changes when backporting some of these concepts. I've reallign it now to use `RequestSpan` module to process creation, storage and retrieval of spans.